crates/symbolicator/src/services/download/gcs.rs

```diff
@@ -260,6 +263,18 @@ impl GcsDownloader {
         }
     }
 
+    pub async fn download_source(
+        &self,
+        file_source: GcsRemoteDif,
+        destination: PathBuf,
+    ) -> Result<Result<DownloadStatus, DownloadError>, Elapsed> {
+        tokio::time::timeout(
+            self.download_timeout,
+            self.download_source_inner(file_source, destination),
+        )
```

Comment:
So I'm questioning if this is the right choice of letting the two timeouts interact.  I commented in detail on https://getsentry.atlassian.net/browse/NATIVE-59 but in short, maybe it was better to leave download_timeout at the top as an absolute maximum time a download is allowed to have.  If so that would remove the need to pass that timeout along through everything.

Reply:
we would still need to pass down these more targeted timeouts through all the structs/calls, even if we keep the top-level timeout, which I agree we should!